Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp

```diff
@@ -962,71 +938,34 @@ ThrowCompletionOr<Object*> default_merge_calendar_fields(VM& vm, Object const& f
     // 1. Let merged be OrdinaryObjectCreate(%Object.prototype%).
     auto merged = Object::create(realm, realm.intrinsics().object_prototype());
 
-    // 2. Let fieldsKeys be ? EnumerableOwnPropertyNames(fields, key).
-    auto fields_keys = TRY(fields.enumerable_own_property_names(Object::PropertyKind::Key));
-
-    // 3. For each element key of fieldsKeys, do
-    for (auto& key : fields_keys) {
-        // a. If key is not "month" or "monthCode", then
-        if (key.as_string().deprecated_string() != vm.names.month.as_string() && key.as_string().deprecated_string() != vm.names.monthCode.as_string()) {
-            auto property_key = MUST(PropertyKey::from_value(vm, key));
-
-            // i. Let propValue be ? Get(fields, key).
-            auto prop_value = TRY(fields.get(property_key));
-
-            // ii. If propValue is not undefined, then
-            if (!prop_value.is_undefined()) {
-                // 1. Perform ! CreateDataPropertyOrThrow(merged, key, propValue).
-                MUST(merged->create_data_property_or_throw(property_key, prop_value));
-            }
-        }
-    }
-
-    // 4. Let additionalFieldsKeys be ? EnumerableOwnPropertyNames(additionalFields, key).
-    auto additional_fields_keys = TRY(additional_fields.enumerable_own_property_names(Object::PropertyKind::Key));
+    // 2. Perform ? CopyDataProperties(merged, fields, « », « undefined »).
+    TRY(merged->copy_data_properties(vm, &fields, {}, { js_undefined() }));
 
-    // IMPLEMENTATION DEFINED: This is an optimization, so we don't have to iterate new_keys three times (worst case), but only once.
-    bool additional_fields_keys_contains_month_or_month_code_property = false;
+    // FIXME: There used to be an optimization which avoided the triple copy if no "month" or
+    //        "monthCode" was present.
```

Comment:
I just added a FIXME since there was an optimization here before, however I'm not noticing a slow down of test 262 nor test js, so perhaps this was premature?